src/navigation/navigationhandler.cpp

```diff
@@ -490,23 +498,42 @@ void NavigationHandler::loadNavigationState(const std::string& filepath) {
     }
 }
 
-void NavigationHandler::setJoystickAxisMapping(int axis,
+void NavigationHandler::setJoystickAxisMapping(const std::string& joystickName, int axis,
                                                JoystickCameraStates::AxisType mapping,
                                             JoystickCameraStates::AxisInvert shouldInvert,
-                                      JoystickCameraStates::AxisNormalize shouldNormalize,
+                                          JoystickCameraStates::JoystickType joystickType,
                                                bool isSticky,
                                                double sensitivity)
 {
     _orbitalNavigator.joystickStates().setAxisMapping(
+        joystickName,
         axis,
         mapping,
         shouldInvert,
-        shouldNormalize,
+        joystickType,
         isSticky,
         sensitivity
     );
 }
 
+void NavigationHandler::setJoystickAxisMappingProperty(const std::string& joystickName,
+                                                       int axis,
+                                                       const std::string& propertyUri,
```

Comment:
Transitively, this `propertyUri` can be `std::string` and then moved into the axis mapping function